### PR TITLE
Removed xfail for physics tests and improved stability

### PR DESCRIPTION
**Pull request by moraaar**
This PR has the same changes done in https://github.com/o3de/o3de/pull/5399, updated to the latest from development branch and resolving conflicts. Its modifications include the following:

- Physics tests use LyTestTools framework, instead of automatedtesting_shared. It's the most recent way to run the tests and more stable.
- Physics Main Optimized tests is a superset of Physics Main tests. Now that Physics Main Optimized have been proved to work it has been made the default set of tests for physics.
- Fixed how MultiplayerEditorSystemComponent.cpp is querying if prefabs is enabled or not.
- Updated `editor_python_test_tools/utils.py` to not close the editor. From @AlexOteiza: "No test should ever close the editor. This code is just wrong"

````
E:\Amazon\GitHub\o3de\AutomatedTesting>..\python\python.cmd -m pytest .\Gem\PythonTests\Physics\TestSuite_Main.py --build-directory=build\windows\bin\profile\
================================================= test session starts =================================================
platform win32 -- Python 3.7.12, pytest-5.3.2, py-1.11.0, pluggy-0.13.1
rootdir: E:\Amazon\GitHub\o3de, inifile: pytest.ini
plugins: mock-2.0.0, timeout-1.3.4, ly-test-tools-1.0.0
collected 86 items

Gem\PythonTests\Physics\TestSuite_Main.py ...xxx.........Xx......x...........................................X.. [ 79%]
.......x..........                                                                                               [100%]

================================================== warnings summary ===================================================
e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333
  e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333: PytestCollectionWarning: cannot collect test class 'TestData' because it has a __init__ constructor (from: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py::TestAutomation)
    class TestData:

e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333
  e:\amazon\github\o3de\tools\lytesttools\ly_test_tools\o3de\editor_test.py:333: PytestCollectionWarning: cannot collect test class 'TestData' because it has a __init__ constructor (from: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py::TestAutomationNoPrefab)
    class TestData:

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================== 80 passed, 6 xfailed, 2 xpassed, 2 warnings in 117.24s (0:01:57) ===========================
````

Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Physics/CMakeLists.txt [hunk 1]
```diff
@@ -12,18 +12,6 @@ if(PAL_TRAIT_BUILD_TESTS_SUPPORTED AND PAL_TRAIT_BUILD_HOST_TOOLS)
...
         PATH ${CMAKE_CURRENT_LIST_DIR}/TestSuite_Main.py
-        RUNTIME_DEPENDENCIES
-            Legacy::Editor
-            AZ::AssetProcessor
-            AutomatedTesting.Assets
-        COMPONENT
-            Physics
-    )
-    ly_add_pytest(
-        NAME AutomatedTesting::PhysicsTests_Main_Optimized
```
Comment (greerdv) [inline AutomatedTesting/Gem/PythonTests/Physics/CMakeLists.txt:23]: What happens to the main optimized test suite with this change? Is it basically replacing the main test suite?

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py [hunk 2]
```diff
@@ -209,10 +215,8 @@ def fail_fast(message=None):
         raises FailFast
         :return: None
         """
-        Report.info("Failing fast. Raising an exception and shutting down the editor.")
         if message:
             Report.info("Fail fast message: {}".format(message))
-        TestHelper.close_editor()
```
Comment (greerdv) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:215]: This seems like a more general change affecting more than just physics. I know you've already gone to a tonne of extra effort resurrecting this PR, but would it be worth splitting this change into a separate PR?

File: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py [hunk 3]
```diff
@@ -3,104 +3,367 @@
...
+
+    class Collider_MultipleSurfaceSlots(EditorSharedTest):
+        from .tests.collider import Collider_MultipleSurfaceSlots as test_module
+
+    class Collider_PxMeshAutoAssignedWhenAddingRenderMeshComponent(EditorSharedTest):
+        from .tests.collider import Collider_PxMeshAutoAssignedWhenAddingRenderMeshComponent as test_module
+        
+    @pytest.mark.xfail(reason="AssertionError: Couldn't find Asset with path: Objects/SphereBot/r0-b_body.azmodel")
+    class Collider_PxMeshAutoAssignedWhenModifyingRenderMeshComponent(EditorSharedTest):
+        from .tests.collider import Collider_PxMeshAutoAssignedWhenModifyingRenderMeshComponent as test_module
```
Comment (moraaar) [inline AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py:142]: This test had the wrong xfail reason. It's failing because it's trying to use an asset the doesn't exist in AutomatedTesting.

This will have to be addressed at later.

File: AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py [hunk 4]
```diff
@@ -3,104 +3,367 @@
...
+    class Physics_UndoRedoWorksOnEntityWithPhysComponents(EditorSharedTest):
+        from .tests import Physics_UndoRedoWorksOnEntityWithPhysComponents as test_module
+
+    @pytest.mark.xfail(reason="AssertionError: Failed to open level: ScriptCanvas_ShapeCast does not exist or is invalid")
+    class ScriptCanvas_ShapeCast(EditorSharedTest):
+        from .tests.script_canvas import ScriptCanvas_ShapeCast as test_module
+
+    @pytest.mark.xfail(reason="AssertionError: Failed to open level: ForceRegion_SliceFileInstantiates does not exist or is invalid")
+    class ForceRegion_SliceFileInstantiates(EditorSharedTest):
+        from .tests.force_region import ForceRegion_SliceFileInstantiates as test_module
```
Comment (moraaar) [inline AutomatedTesting/Gem/PythonTests/Physics/TestSuite_Main.py:369]: All these tests are failing and I've added the specific reason on each one. 

Before it was the whole `class TestAutomationNoPrefab` the one that had xfail, so all the tests could fail and not complain. I've changed it so each tests now has the specific reason why it fails.

This will have to be addressed at later.

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp [hunk 5]
```diff
@@ -272,7 +273,9 @@ namespace Multiplayer
         auto prefabEditorEntityOwnershipInterface = AZ::Interface<AzToolsFramework::PrefabEditorEntityOwnershipInterface>::Get();
         if (!prefabEditorEntityOwnershipInterface)
         {
-            AZ_Error("MultiplayerEditor", prefabEditorEntityOwnershipInterface != nullptr, "PrefabEditorEntityOwnershipInterface unavailable");
```
Comment (greerdv) [inline Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp:275]: Could you give some context how these changes are related to the rest of the PR please? I'm not immediately clear on that.

**Review by greerdv — COMMENTED** [review]
I'm a little bit concerned that there are a few distinct things going on here. For example, it looks like the main suite is getting replaced by the main optimized suite, which isn't really called out in the title or the description.

File: AutomatedTesting/Gem/PythonTests/Physics/CMakeLists.txt (hunk 1 above)
Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/Physics/CMakeLists.txt:23]: Yes. That's what Alex's PR did, get rid of main and use main optimized instead (moved main optimized -> main).

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py (hunk 2 above)
Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:215]: From Alex's comment about that line:

Aaron: Is it safe to remove this? are other tests relying on this?
Alex: No test should ever close the editor. This code is just wrong

I included sig-testing here, they can take a look.

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp (hunk 5 above)
Comment (moraaar) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp:275]: It's the changes that Alex did, I don't have context, but it seems like he is using now the correct boolean to check if prefabs are enabled or not. I've included sig-networking so they can take a look.

**Comment by moraaar** [comment]
> I'm a little bit concerned that there are a few distinct things going on here. For example, it looks like the main suite is getting replaced by the main optimized suite, which isn't really called out in the title or the description.

I've updated the description with all the changes @AlexOteiza did.

**Review by SergeyAMZN — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py (hunk 2 above)
Comment (evanchia-ly-sdets) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:215]: Looks sane to me. The only concern is if some test is using this function directly (which it shouldn't) so a main suite pass should be good enough evidence.

Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:215]: Thanks @evanchia-ly-sdets, AR for this PR looks green: https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8066/2/pipeline 

Is there anything else I should try to be safe?

**Review by evanchia-ly-sdets — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py (hunk 2 above)
Comment (evanchia-ly-sdets) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py:215]: lgtm

**Review by Kadino — APPROVED** [review]

**Comment by Kadino** [comment]
Merging this approved and passing PR as the remainder of the suite is seeing intermittent failures

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp (hunk 5 above)
Comment (AlexOteiza) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp:275]: Sorry I'm quite late for this, but can give some insight.
This always errored when having prefabs disabled, which is not correct and made the test to fail.

If we are in prefab enabled mode and the interface does not exist, then that is an error.
But in slice mode, is expected that this interface should not exist and that should *not* be an error.

In the future if you need more info on anything, just message me on discord :)
